File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestInteger.js
print("getInteger")
if(typeof getInteger !== 'undefined')
	throw "getInteger already defined!"
getInteger = callback("getInteger")
getIntegerType = typeof getInteger
print(getIntegerType)
print(getInteger)
if(getIntegerType !== 'number')
	throw "getInteger not number!"
callback("setInteger",getInteger)

print("getIntegerVector")
if(typeof getIntegerVector !== 'undefined')
	throw "getIntegerVector already defined!"
getIntegerVector = callback("getIntegerVector")
getIntegerVectorType = typeof getIntegerVector[0]
print(getIntegerVectorType)
print(getIntegerVector)
if(getIntegerVectorType !== 'number')
	throw "getIntegerVector not number!"
callback("setIntegerVector",getIntegerVector)

print("getIntegerVectorAsList")
if(typeof getIntegerVectorAsList !== 'undefined')
	throw "getIntegerVectorAsList already defined!"
getIntegerVectorAsList = callback("getIntegerVectorAsList")
getIntegerVectorAsListType = typeof getIntegerVectorAsList[0]
print(getIntegerVectorAsListType)
print(getIntegerVectorAsList)
if(getIntegerVectorAsListType !== 'number')
	throw "getIntegerVectorAsList not number!"
callback("setIntegerVectorAsList",getIntegerVectorAsList)

print("getIntegerMatrix")
if(typeof getIntegerMatrix !== 'undefined')
	throw "getIntegerMatrix already defined!"
getIntegerMatrix = callback("getIntegerMatrix")
getIntegerMatrixType = typeof getIntegerMatrix[0][0]
print(getIntegerMatrixType)
print(getIntegerMatrix)
if(getIntegerMatrixType !== 'number')
	throw "getIntegerMatrix not number!"
callback("setIntegerMatrix",getIntegerMatrix)

print("getIntegerMatrixAsList")
if(typeof getIntegerMatrixAsList !== 'undefined')
	throw "getIntegerMatrixAsList already defined!"
getIntegerMatrixAsList = callback("getIntegerMatrixAsList")
getIntegerMatrixAsListType = typeof getIntegerMatrixAsList[0][0]
print(getIntegerMatrixAsListType)
print(getIntegerMatrixAsList)
if(getIntegerMatrixAsListType !== 'number')
	throw "getIntegerMatrixAsList not number!"
callback("setIntegerMatrixAsList",getIntegerMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-ruby-contract/src/test/java/de/invesdwin/context/ruby/tests/callback/ParametersAndReturnsTestInteger.rb
puts("getInteger")
unless (defined?(getInteger)).nil? || getInteger.nil?
	raise "getInteger already defined!"
end
getInteger = callback("getInteger")
getIntegerType = getInteger.class
puts(getIntegerType)
puts(getInteger)
if(getIntegerType != Integer)
	raise "getInteger not Integer!"
end
callback("setInteger", getInteger)

puts("getIntegerVector")
unless (defined?(getIntegerVector)).nil? || getIntegerVector.nil?
	raise "getIntegerVector already defined!"
end
getIntegerVector = callback("getIntegerVector")
getIntegerVectorType = getIntegerVector[0].class
puts(getIntegerVectorType)
puts(getIntegerVector)
if(getIntegerVectorType != Integer)
	raise "getIntegerVector not Integer!"
end
callback("setIntegerVector", getIntegerVector)

puts("getIntegerVectorAsList")
unless (defined?(getIntegerVectorAsList)).nil? || getIntegerVectorAsList.nil?
	raise "getIntegerVectorAsList already defined!"
end
getIntegerVectorAsList = callback("getIntegerVectorAsList")
getIntegerVectorAsListType = getIntegerVectorAsList[0].class
puts(getIntegerVectorAsListType)
puts(getIntegerVectorAsList)
if(getIntegerVectorAsListType != Integer)
	raise "getIntegerVectorAsList not Integer!"
end
callback("setIntegerVectorAsList", getIntegerVectorAsList)

puts("getIntegerMatrix")
unless (defined?(getIntegerMatrix)).nil? || getIntegerMatrix.nil?
	raise "getIntegerMatrix already defined!"
end
getIntegerMatrix = callback("getIntegerMatrix")
getIntegerMatrixType = getIntegerMatrix[0][0].class
puts(getIntegerMatrixType)
puts(getIntegerMatrix)
if(getIntegerMatrixType != Integer)
	raise "getIntegerMatrix not Integer!"
end
callback("setIntegerMatrix", getIntegerMatrix)

puts("getIntegerMatrixAsList")
unless (defined?(getIntegerMatrixAsList)).nil? || getIntegerMatrixAsList.nil?
	raise "getIntegerMatrixAsList already defined!"
end
getIntegerMatrixAsList = callback("getIntegerMatrixAsList")
getIntegerMatrixAsListType = getIntegerMatrixAsList[0][0].class
puts(getIntegerMatrixAsListType)
puts(getIntegerMatrixAsList)
if(getIntegerMatrixAsListType != Integer)
	raise "getIntegerMatrixAsList not Integer!"
end
callback("setIntegerMatrixAsList", getIntegerMatrixAsList)
